### DISCUSSION TITLE
Request for Guidance - Embedding in another page.

### DISCUSSION THREAD

**Discussion by the-maldridge** [drakkan/sftpgo#1084] [General]
I'm looking at a use case that involves replacing another file manager that's currently embedded in a larger web app.  The experience provided by sftpgo is much better, but I'd probably still need to provide an embedded experience.  Is there any existing path for either embedding the share view, or a way for my external app to get per-file authenticated links to hand back to users?

Thanks in advance!

**Comment by drakkan** [comment]
Really? I think the web client UI needs a lot of improvement, for example compare it with Nextcloud. I tried to improve it myself but failed. I will hire a web developer once I get enough sponsorships (if that ever happens).

There is no supported way to embed the web UI within another application. You can use the REST API and develop a web UI yourself.
 
Some time ago I provided to a one-time sponsor a sample program to externally generate a valid cookie for the WebClient, I don't know if it still works with recent versions of SFTPGo. Maybe if he reads this thread he could share that sample
